qutip/nonmarkov/bofin_baths.py

```diff
@@ -0,0 +1,729 @@
+        exponents within the bath) of the corresponding "-" bath exponent.
+        For exponents of type "-" it gives the offset of the corresponding
+        "+" exponent.
+
+    tag : optional, str, tuple or any other object
+        A label for the exponent (often the name of the bath). It
+        defaults to None.
+
+    Attributes
+    ----------
+
+    All of the parameters are available as attributes.
+    """
+    types = enum.Enum("ExponentType", ["R", "I", "RI", "+", "-"])
+
+    def _check_ck2(self, type, ck2):
+        if type == self.types["RI"]:
+            if ck2 is None:
+                raise ValueError("RI bath exponents require ck2")
+        else:
+            if ck2 is not None:
+                raise ValueError(
+                    "Second co-efficient (ck2) should only be specified for RI"
+                    " bath exponents"
+                )
+
+    def _check_sigma_bar_k_offset(self, type, offset):
+        if type in (self.types["+"], self.types["-"]):
+            if offset is None:
+                raise ValueError(
+                    "+ and - bath exponents require sigma_bar_k_offset"
+                )
+        else:
+            if offset is not None:
+                raise ValueError(
+                    "Offset of sigma bar (sigma_bar_k_offset) should only be"
+                    " specified for + and - bath exponents"
+                )
+
+    def __init__(
+            self, type, dim, Q, ck, vk, ck2=None, sigma_bar_k_offset=None,
+            tag=None,
+    ):
+        if not isinstance(type, self.types):
+            type = self.types[type]
+        self._check_ck2(type, ck2)
+        self._check_sigma_bar_k_offset(type, sigma_bar_k_offset)
+        self.type = type
+        self.dim = dim
+        self.Q = Q
+        self.ck = ck
+        self.vk = vk
+        self.ck2 = ck2
+        self.sigma_bar_k_offset = sigma_bar_k_offset
+        self.tag = tag
+
+    def __repr__(self):
+        dims = getattr(self.Q, "dims", None)
+        return (
+            f"<{self.__class__.__name__} type={self.type.name}"
+            f" dim={self.dim!r}"
+            f" Q.dims={dims!r}"
+            f" ck={self.ck!r} vk={self.vk!r} ck2={self.ck2!r}"
+            f" sigma_bar_k_offset={self.sigma_bar_k_offset!r}"
+            f" tag={self.tag!r}>"
+        )
+
+
+class Bath:
+    """
+    Represents a list of bath expansion exponents.
+
+    Parameters
+    ----------
+    exponents : list of BathExponent
+        The exponents of the correlation function describing the bath.
+
+    Attributes
+    ----------
+
+    All of the parameters are available as attributes.
+    """
+    def __init__(self, exponents):
+        self.exponents = exponents
+
+
+class BosonicBath(Bath):
+    """
+    A helper class for constructing a bosonic bath from the expansion
+    coefficients and frequencies for the real and imaginary parts of
+    the bath correlation function.
```

Comment:
expansion is vague, it could be a taylor expansion, fourrier expansion, since nothing is said, it's up to the reader to guess... If you can add a reference to the paper, it would be great.
Otherwise, the equation should be found somewhere in the documentation, just something like: 
`correlation(t) = sum( ck_real * np.exp(-vk_real*t) +1j  ck_imag * np.exp(-vk_imag*t) )`
would help.

Reply:
The paper isn't published yet, but I've added equations to the docstrings.